build failed while processing submodule arrow-cmake

> Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.com/docs/en/development/build/
```
[root@localhost ClickHouse]# git clone --recursive --shallow-submodules https://github.com/ClickHouse/ClickHouse.git
[root@localhost ClickHouse]# git diff
[root@localhost ClickHouse]# git status
On branch master
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean
```

**Operating system**

> OS kind or distribution, specific version/release, non-standard kernel if any. If you are trying to build inside virtual machine, please mention it too.
```
[root@localhost ClickHouse]# uname -a
Linux localhost.localdomain 4.19.90-52.15.v2207.ky10.aarch64 #1 SMP Thu Nov 24 15:59:41 CST 2022 aarch64 aarch64 aarch64 GNU/Linux
[root@localhost ClickHouse]# lscpu
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              1
Core(s) per socket:              32
Socket(s):                       2
NUMA node(s):                    4
Vendor ID:                       ARM
Model:                           2
Model name:                      Cortex-A72
Stepping:                        r0p2
BogoMIPS:                        100.00
L1d cache:                       2 MiB
L1i cache:                       3 MiB
L2 cache:                        16 MiB
L3 cache:                        64 MiB
NUMA node0 CPU(s):               0-15
NUMA node1 CPU(s):               16-31
NUMA node2 CPU(s):               32-47
NUMA node3 CPU(s):               48-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Vulnerable
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
```

**Cmake version**
```
[root@localhost ClickHouse]# cmake --version
cmake version 3.26.0-rc1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Ninja version**
```
[root@localhost ClickHouse]# ninja --version
1.12.0.git
```

**Compiler name and version**
```
[root@localhost ClickHouse]# clang --version
clang version 15.0.7
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/llvm/bin


[root@localhost ClickHouse]# clang++ --version
clang version 15.0.7
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/llvm/bin


[root@localhost ClickHouse]# gcc --version
gcc (GCC) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


[root@localhost ClickHouse]# g++ --version
g++ (GCC) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Full cmake and/or ninja output**
**Cmake:**
```
[root@localhost build]# cmake ..
-- The C compiler identification is Clang 15.0.7
-- The CXX compiler identification is Clang 15.0.7
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /usr/local/llvm/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Cross-compiling for target: 
-- Using compiler:
clang version 15.0.7
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/llvm/bin
-- Using linker: /usr/local/llvm/bin/ld.lld
-- Using archiver: /usr/local/llvm/bin/llvm-ar
-- Using ranlib: /usr/local/llvm/bin/llvm-ranlib
-- Using install-name-tool: /usr/local/llvm/bin/llvm-install-name-tool
-- Using objcopy: /usr/local/llvm/bin/llvm-objcopy
-- Using strip: /usr/local/llvm/bin/llvm-strip
-- Using ccache: /usr/bin/ccache (version 3.3.4)
-- Found Git: /usr/bin/git (found version "2.27.0") 
-- Git HEAD commit hash: 0db943e596209f877b0d89b9d8913bbe42f3721d
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
-- CMAKE_BUILD_TYPE is not set, set to default = RelWithDebInfo
-- CMAKE_BUILD_TYPE: RelWithDebInfo
-- Performing Test HAS_RESERVED_IDENTIFIER
-- Performing Test HAS_RESERVED_IDENTIFIER - Success
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE - Success
-- Performing Test HAS_SUGGEST_OVERRIDE
-- Performing Test HAS_SUGGEST_OVERRIDE - Success
-- Performing Test HAS_USE_CTOR_HOMING
-- Performing Test HAS_USE_CTOR_HOMING - Success
-- Performing Test SUPPORTS_CXXFLAG_no_enum_constexpr_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_enum_constexpr_conversion - Failed
-- Performing Test SUPPORTS_CFLAG_no_enum_constexpr_conversion
-- Performing Test SUPPORTS_CFLAG_no_enum_constexpr_conversion - Failed
-- Flag -Wno-enum-constexpr-conversion is unsupported
-- Flag -Wno-enum-constexpr-conversion is unsupported
-- Adding .gdb-index via --gdb-index linker option.
-- No official build: A checksum hash will not be added to the clickhouse executable
-- Default libraries: -nodefaultlibs -lgcc  -lc -lm -lrt -lpthread -ldl
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_WARNING_UNUSED_BUT_SET_VARIABLE
-- Performing Test HAVE_WARNING_UNUSED_BUT_SET_VARIABLE - Success
-- Performing Test HAVE_WARNING_MISSING_ATTRIBUTES
-- Performing Test HAVE_WARNING_MISSING_ATTRIBUTES - Failed
-- Performing Test HAVE_WARNING_MAYBE_UNINITIALIZED
-- Performing Test HAVE_WARNING_MAYBE_UNINITIALIZED - Failed
-- Performing Test HAVE_WARNING_REGISTER
-- Performing Test HAVE_WARNING_REGISTER - Success
-- Using libunwind: unwind
-- Using exception handler: unwind
-- Looking for sys/random.h
-- Looking for sys/random.h - found
-- Performing Test SUPPORTS_CXXFLAG_no_unused_command_line_argument
-- Performing Test SUPPORTS_CXXFLAG_no_unused_command_line_argument - Success
-- Performing Test SUPPORTS_CXXFLAG_no_unused_but_set_variable
-- Performing Test SUPPORTS_CXXFLAG_no_unused_but_set_variable - Success
-- Performing Test SUPPORTS_CXXFLAG_no_builtin_requires_header
-- Performing Test SUPPORTS_CXXFLAG_no_builtin_requires_header - Success
-- Some symbols from glibc will be replaced for compatibility
-- Unit tests are enabled
-- Building for: Linux aarch64 
-- Adding contrib module boringssl (configuring with boringssl-cmake)
-- Adding contrib module miniselect (configuring with miniselect-cmake)
-- Adding contrib module pdqsort (configuring with pdqsort-cmake)
-- Adding contrib module crc32-vpmsum (configuring with crc32-vpmsum-cmake)
-- crc32-vpmsum library is only supported on ppc64le
-- Adding contrib module sparsehash-c11 (configuring with sparsehash-c11-cmake)
-- Adding contrib module abseil-cpp (configuring with abseil-cpp-cmake)
-- Adding contrib module magic_enum (configuring with magic-enum-cmake)
-- Adding contrib module boost (configuring with boost-cmake)
-- Adding contrib module cctz (configuring with cctz-cmake)
-- Packaging with tzdata version: 2022f
-- Adding contrib module consistent-hashing (configuring with consistent-hashing)
-- Adding contrib module dragonbox (configuring with dragonbox-cmake)
-- Adding contrib module vectorscan (configuring with vectorscan-cmake)
-- Adding contrib module jemalloc (configuring with jemalloc-cmake)
-- jemalloc malloc_conf: percpu_arena:percpu,oversize_threshold:0,muzzy_decay_ms:5000,dirty_decay_ms:5000
-- Adding contrib module libcpuid (configuring with libcpuid-cmake)
-- Not using cpuid
-- Adding contrib module libdivide-cmake (configuring with libdivide-cmake)
-- Adding contrib module libmetrohash (configuring with libmetrohash)
-- Adding contrib module lz4 (configuring with lz4-cmake)
-- Adding contrib module murmurhash (configuring with murmurhash)
-- Adding contrib module replxx (configuring with replxx-cmake)
-- Adding contrib module unixodbc (configuring with unixodbc-cmake)
-- Adding contrib module nanodbc (configuring with nanodbc-cmake)
-- Adding contrib module capnproto (configuring with capnproto-cmake)
-- Adding contrib module yaml-cpp (configuring with yaml-cpp-cmake)
-- Adding contrib module re2 (configuring with re2-cmake)
-- Adding contrib module xz (configuring with xz-cmake)
-- Adding contrib module brotli (configuring with brotli-cmake)
-- Adding contrib module double-conversion (configuring with double-conversion-cmake)
-- Adding contrib module croaring (configuring with croaring-cmake)
-- Adding contrib module zstd (configuring with zstd-cmake)
-- ZSTD VERSION 1.5.0
-- Adding contrib module zlib-ng (configuring with zlib-ng-cmake)
-- Adding contrib module bzip2 (configuring with bzip2-cmake)
-- Adding contrib module minizip-ng (configuring with minizip-ng-cmake)
-- Adding contrib module snappy (configuring with snappy-cmake)
-- Adding contrib module rocksdb (configuring with rocksdb-cmake)
-- Performing Test HAS_ARMV8_CRC
-- Performing Test HAS_ARMV8_CRC - Success
--  HAS_ARMV8_CRC yes
-- Performing Test HAVE_FALLOCATE
-- Performing Test HAVE_FALLOCATE - Success
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE - Failed
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP - Success
-- Looking for malloc_usable_size
-- Looking for malloc_usable_size - not found
-- Adding contrib module thrift (configuring with thrift-cmake)
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/un.h
-- Looking for sys/un.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for sys/poll.h
-- Looking for sys/poll.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sched.h
-- Looking for sched.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Performing Test HAVE_AF_UNIX_H
-- Performing Test HAVE_AF_UNIX_H - Failed
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for sched_get_priority_max
-- Looking for sched_get_priority_max - found
-- Looking for sched_get_priority_min
-- Looking for sched_get_priority_min - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Success
-- Adding contrib module arrow (configuring with arrow-cmake)
-- Looking for strtof_l
-- Looking for strtof_l - found
-- Looking for strtoull_l
-- Looking for strtoull_l - found
-- Using toolchain file: /home/ph/build/ClickHouse/cmake/linux/toolchain-aarch64.cmake.
-- CMAKE_CXX_FLAGS:  --gcc-toolchain=/home/ph/build/ClickHouse/cmake/linux/../../contrib/sysroot/linux-aarch64 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument -fasynchronous-unwind-tables -ffile-prefix-map=/home/ph/build/ClickHouse=. -falign-functions=32 -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w
fatal: No tags can describe 'eb3f827948241ce0e701516f16cd67324802bce9'.
Try --always, or create some tags.
CMake Warning at contrib/flatbuffers/CMake/Version.cmake:22 (message):
  git describe failed with exit code: 128
Call Stack (most recent call first):
  contrib/flatbuffers/CMakeLists.txt:562 (include)


-- Proceeding with version: 2.0.0.0
-- Adding contrib module avro (configuring with avro-cmake)
-- Adding contrib module protobuf (configuring with protobuf-cmake)
-- Adding contrib module openldap (configuring with openldap-cmake)
-- Adding contrib module grpc (configuring with grpc-cmake)
-- Not using gRPC
-- Adding contrib module msgpack-c (configuring with msgpack-c-cmake)
-- Adding contrib module corrosion (configuring with corrosion-cmake)
No such file or directory
CMake Warning at contrib/corrosion/cmake/FindRust.cmake:176 (message):
  The rustc executable was not found.  Rust not installed or ~/.cargo/bin not
  added to path? Aborting further actions of find_package(Rust).
Call Stack (most recent call first):
  contrib/corrosion-cmake/CMakeLists.txt:8 (find_package)


-- Not using rust
-- Adding contrib module wyhash (configuring with wyhash-cmake)
-- Adding contrib module cityhash102 (configuring with cityhash102)
-- Adding contrib module libfarmhash (configuring with libfarmhash)
-- Adding contrib module icu (configuring with icu-cmake)
-- Adding contrib module h3 (configuring with h3-cmake)
-- Adding contrib module mariadb-connector-c (configuring with mariadb-connector-c-cmake)
-- Adding contrib module googletest (configuring with googletest-cmake)
-- Adding contrib module llvm-project (configuring with llvm-project-cmake)
-- Not using LLVM
-- Adding contrib module llvm-project (configuring with libfuzzer-cmake)
-- Adding contrib module llvm-project (configuring with gwpasan-cmake)
-- Adding contrib module libxml2 (configuring with libxml2-cmake)
-- Adding contrib module aws;aws-c-auth;aws-c-cal;aws-c-common;aws-c-compression;aws-c-event-stream;aws-c-http;aws-c-io;aws-c-mqtt;aws-c-s3;aws-c-sdkutils;aws-s2n-tls;aws-checksums;aws-crt-cpp;aws-cmake (configuring with aws-cmake)
-- Performing Test AWS_HAVE_WINAPI_DESKTOP
-- Performing Test AWS_HAVE_WINAPI_DESKTOP - Failed
-- Performing Test AWS_ARCH_INTEL
-- Performing Test AWS_ARCH_INTEL - Failed
-- Performing Test AWS_ARCH_ARM64
-- Performing Test AWS_ARCH_ARM64 - Success
-- Performing Test AWS_ARCH_ARM32
-- Performing Test AWS_ARCH_ARM32 - Failed
-- Performing Test AWS_HAVE_GCC_INLINE_ASM
-- Performing Test AWS_HAVE_GCC_INLINE_ASM - Success
-- Performing Test AWS_HAVE_AUXV
-- Performing Test AWS_HAVE_AUXV - Success
-- Performing Test AWS_HAVE_EXECINFO
-- Performing Test AWS_HAVE_EXECINFO - Failed
-- Performing Test AWS_HAVE_LINUX_IF_LINK_H
-- Performing Test AWS_HAVE_LINUX_IF_LINK_H - Success
-- Performing Test HAVE_M_AVX2_FLAG
-- Performing Test HAVE_M_AVX2_FLAG - Success
-- Performing Test HAVE_AVX2_INTRINSICS
-- Performing Test HAVE_AVX2_INTRINSICS - Failed
-- Performing Test HAVE_MM256_EXTRACT_EPI64
-- Performing Test HAVE_MM256_EXTRACT_EPI64 - Failed
-- Looking for pthread_attr_setaffinity_np
-- Looking for pthread_attr_setaffinity_np - found
-- Performing Test PTHREAD_SETNAME_TAKES_2ARGS
-- Performing Test PTHREAD_SETNAME_TAKES_2ARGS - Success
-- Adding contrib module base64 (configuring with base64-cmake)
-- Adding contrib module simdjson (configuring with simdjson-cmake)
-- Adding contrib module rapidjson (configuring with rapidjson-cmake)
-- Adding contrib module fastops (configuring with fastops-cmake)
-- Not using fastops
-- Adding contrib module libuv (configuring with libuv-cmake)
-- Adding contrib module liburing (configuring with liburing-cmake)
-- Adding contrib module AMQP-CPP (configuring with amqpcpp-cmake)
-- Adding contrib module cassandra (configuring with cassandra-cmake)
-- Adding contrib module curl (configuring with curl-cmake)
-- Adding contrib module azure (configuring with azure-cmake)
By default, if no option is selected, on POSIX, libcurl transport adapter is used.
-- Adding contrib module sentry-native (configuring with sentry-native-cmake)
-- Not using sentry
-- Adding contrib module fmtlib (configuring with fmtlib-cmake)
-- Adding contrib module krb5 (configuring with krb5-cmake)
-- Adding contrib module cyrus-sasl (configuring with cyrus-sasl-cmake)
-- Adding contrib module libgsasl (configuring with libgsasl-cmake)
-- Adding contrib module librdkafka (configuring with librdkafka-cmake)
-- librdkafka with SASL support
-- librdkafka with SSL support
-- Adding contrib module nats-io (configuring with nats-io-cmake)
-- Adding contrib module libhdfs3 (configuring with libhdfs3-cmake)
-- Not using hdfs
-- Adding contrib module hive-metastore (configuring with hive-metastore-cmake)
-- Not using hive
-- Adding contrib module cppkafka (configuring with cppkafka-cmake)
-- Adding contrib module libpqxx (configuring with libpqxx-cmake)
-- Adding contrib module libpq (configuring with libpq-cmake)
-- Adding contrib module NuRaft (configuring with nuraft-cmake)
-- Adding contrib module fast_float (configuring with fast_float-cmake)
-- Adding contrib module datasketches-cpp (configuring with datasketches-cpp-cmake)
-- Adding contrib module hashidsxx (configuring with hashidsxx-cmake)
-- Adding contrib module libstemmer_c (configuring with libstemmer-c-cmake)
-- Adding contrib module wordnet-blast (configuring with wordnet-blast-cmake)
-- Adding contrib module lemmagen-c (configuring with lemmagen-c-cmake)
-- Adding contrib module nlp-data (configuring with nlp-data-cmake)
-- Adding contrib module cld2 (configuring with cld2-cmake)
-- Adding contrib module sqlite-amalgamation (configuring with sqlite-cmake)
-- Adding contrib module s2geometry (configuring with s2geometry-cmake)
-- Adding contrib module c-ares (configuring with c-ares-cmake)
-- Adding contrib module qpl (configuring with qpl-cmake)
-- Not using QPL
-- Adding contrib module morton-nd (configuring with morton-nd-cmake)
-- Adding contrib module annoy (configuring with annoy-cmake)
-- Not using annoy
-- Adding contrib module xxHash (configuring with xxHash-cmake)
-- Adding contrib module google-benchmark (configuring with google-benchmark-cmake)
-- Adding contrib module ulid-c (configuring with ulid-c-cmake)
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CXXFLAG_everything
-- Performing Test SUPPORTS_CXXFLAG_everything - Success
-- Performing Test SUPPORTS_CFLAG_everything
-- Performing Test SUPPORTS_CFLAG_everything - Success
-- Performing Test SUPPORTS_CXXFLAG_pedantic
-- Performing Test SUPPORTS_CXXFLAG_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_pedantic
-- Performing Test SUPPORTS_CFLAG_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx20_compat
-- Performing Test SUPPORTS_CXXFLAG_no_cxx20_compat - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx20_compat
-- Performing Test SUPPORTS_CFLAG_no_cxx20_compat - Success
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_implicit_int_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_implicit_int_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_implicit_int_conversion
-- Performing Test SUPPORTS_CFLAG_no_implicit_int_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_implicit_int_float_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_implicit_int_float_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_implicit_int_float_conversion
-- Performing Test SUPPORTS_CFLAG_no_implicit_int_float_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CFLAG_no_double_promotion
-- Performing Test SUPPORTS_CFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CFLAG_no_float_equal
-- Performing Test SUPPORTS_CFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CFLAG_no_global_constructors
-- Performing Test SUPPORTS_CFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CXXFLAG_no_padded
-- Performing Test SUPPORTS_CXXFLAG_no_padded - Success
-- Performing Test SUPPORTS_CFLAG_no_padded
-- Performing Test SUPPORTS_CFLAG_no_padded - Success
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CFLAG_no_switch_enum
-- Performing Test SUPPORTS_CFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CFLAG_no_unused_template
-- Performing Test SUPPORTS_CFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_vla
-- Performing Test SUPPORTS_CXXFLAG_no_vla - Success
-- Performing Test SUPPORTS_CFLAG_no_vla
-- Performing Test SUPPORTS_CFLAG_no_vla - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables - Success
-- Performing Test SUPPORTS_CXXFLAG_no_thread_safety_negative
-- Performing Test SUPPORTS_CXXFLAG_no_thread_safety_negative - Success
-- Performing Test SUPPORTS_CFLAG_no_thread_safety_negative
-- Performing Test SUPPORTS_CFLAG_no_thread_safety_negative - Success
-- Flag -Wno-enum-constexpr-conversion is unsupported
-- Flag -Wno-enum-constexpr-conversion is unsupported
-- compiler C   = /usr/local/llvm/bin/clang  --gcc-toolchain=/home/ph/build/ClickHouse/cmake/linux/../../contrib/sysroot/linux-aarch64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing  -gdwarf-aranges -pipe -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument -fasynchronous-unwind-tables -ffile-prefix-map=/home/ph/build/ClickHouse=. -falign-functions=32  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4 
-- compiler CXX = /usr/local/llvm/bin/clang++  --gcc-toolchain=/home/ph/build/ClickHouse/cmake/linux/../../contrib/sysroot/linux-aarch64 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument -fasynchronous-unwind-tables -ffile-prefix-map=/home/ph/build/ClickHouse=. -falign-functions=32 -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4 
-- LINKER_FLAGS =  --ld-path=/home/ph/build/ClickHouse/build/ld.lld -rdynamic -Wl,--gdb-index -Wl,--build-id=sha1 
-- Using Poco::Crypto
-- Using Poco::Data::ODBC
-- /home/ph/build/ClickHouse/src: Have 63920 megabytes of memory.
        Limiting concurrent linkers jobs to 18 and compiler jobs to 25 (system has 64 logical cores)
-- Will build ClickHouse 23.2.1.1 revision 54471 
-- ClickHouse modes:
-- Server mode: ON
-- Client mode: ON
-- Local mode: ON
-- Self-extracting executable: OFF
-- Benchmark mode: ON
-- Extract from config mode: ON
-- Compressor mode: ON
-- Copier mode: ON
-- Format mode: ON
-- Obfuscator mode: ON
-- ODBC bridge mode: ON
-- Library bridge mode: ON
-- ClickHouse install: ON
-- ClickHouse git-import: ON
-- ClickHouse keeper mode: ON
-- ClickHouse keeper-converter mode: ON
-- Clickhouse disks mode: ON
-- ClickHouse su: ON
-- bash_completion will be written to /usr/local/share/bash-completion/completions
-- Target check already exists
-- /home/ph/build/ClickHouse/utils: Have 63920 megabytes of memory.
        Limiting concurrent linkers jobs to 18 and compiler jobs to OFF (system has 64 logical cores)
-- Configuring done (10.5s)
-- Generating done (2.8s)
-- Build files have been written to: /home/ph/build/ClickHouse/build
```

**Ninja:**
```
[root@localhost build]# ninja
[0/2] Re-checking globbed directories...
[3066/9136] Generating orc_proto.pb.h, orc_proto.pb.cc
FAILED: contrib/arrow-cmake/orc_proto.pb.h contrib/arrow-cmake/orc_proto.pb.cc /home/ph/build/ClickHouse/build/contrib/arrow-cmake/orc_proto.pb.h /home/ph/build/ClickHouse/build/contrib/arrow-cmake/orc_proto.pb.cc 
cd /home/ph/build/ClickHouse/build/contrib/arrow-cmake && /home/ph/build/ClickHouse/build/contrib/protobuf-cmake/protoc -I /home/ph/build/ClickHouse/contrib/orc/c++/../proto --cpp_out="/home/ph/build/ClickHouse/build/contrib/arrow-cmake" /home/ph/build/ClickHouse/contrib/orc/c++/../proto/orc_proto.proto
[3131/9136] Building CXX object contrib/icu-cmake/CMakeFiles/_icui18n.dir/__/icu/icu4c/source/i18n/number_scientific.cpp.o
ninja: build stopped: subcommand failed.
```

According to the error message, I ran the command and this is output:
```
[root@localhost build]# /home/ph/build/ClickHouse/build/contrib/protobuf-cmake/protoc -I /home/ph/build/ClickHouse/contrib/orc/c++/../proto --cpp_out="/home/ph/build/ClickHouse/build/contrib/arrow-cmake"
Illegal instruction (core dumped)
```
How to solve it?

--- a/cmake/cpu_features.cmake
+++ b/cmake/cpu_features.cmake
@@ -60,7 +60,7 @@ elseif (ARCH_AARCH64)
         set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument")
     endif ()
 
-    # Quick-n-dirty check: The build generates and executes intermediate binaries, e.g. protoc and llvm-tablegen. If we build on ARM for ARM
+    # Best-effort check: The build generates and executes intermediate binaries, e.g. protoc and llvm-tablegen. If we build on ARM for ARM
     # and the build machine is too old, i.e. doesn't satisfy above modern profile, then these intermediate binaries will not run (dump
     # SIGILL). Even if they could run, the build machine wouldn't be able to run the ClickHouse binary. In that case, suggest to run the
     # build with the compat profile.
@@ -119,7 +119,7 @@ elseif (ARCH_AMD64)
         SET(ENABLE_AVX512_FOR_SPEC_OP 0)
     endif()
 
-    # Same quick-n-dirty check for x86 as above for ARM, see above.
+    # Same best-effort check for x86 as above for ARM, see above.
     if (OS_LINUX AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64|x86_64" AND NOT NO_SSE3_OR_HIGHER)
         # Test for flags in standard profile but not in NO_SSE3_OR_HIGHER profile.
         # /proc/cpuid for Intel Xeon 8124: "fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse

--- a/cmake/cpu_features.cmake
+++ b/cmake/cpu_features.cmake
@@ -65,10 +65,10 @@ elseif (ARCH_AARCH64)
     # SIGILL). Even if they could run, the build machine wouldn't be able to run the ClickHouse binary. In that case, suggest to run the
     # build with the compat profile.
     if (OS_LINUX AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)" AND NOT NO_ARMV81_OR_HIGHER)
-        # CPU features in /proc/cpuinfo and compiler flags don't align :( ... pick some obvious flags contained in modern but not legacy
-        # profile (full Graviton 3 /proc/cpuinfo is "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt
-        # fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs paca pacg dcpodp svei8mm svebf16 i8mm bf16
-        # dgh rng")
+        # CPU features in /proc/cpuinfo and compiler flags don't align :( ... pick some obvious flags contained in the modern but not in the
+        # legacy profile (full Graviton 3 /proc/cpuinfo is "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm
+        # jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs paca pacg dcpodp svei8mm svebf16 i8mm
+        # bf16 dgh rng")
         execute_process(
             COMMAND grep -P "^(?=.*atomic)(?=.*ssbs)" /proc/cpuinfo
             OUTPUT_VARIABLE FLAGS)
@@ -119,7 +119,7 @@ elseif (ARCH_AMD64)
         SET(ENABLE_AVX512_FOR_SPEC_OP 0)
     endif()
 
-    # Same best-effort check for x86 as above for ARM, see above.
+    # Same best-effort check for x86 as above for ARM.
     if (OS_LINUX AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64|x86_64" AND NOT NO_SSE3_OR_HIGHER)
         # Test for flags in standard profile but not in NO_SSE3_OR_HIGHER profile.
         # /proc/cpuid for Intel Xeon 8124: "fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse

--- a/cmake/cpu_features.cmake
+++ b/cmake/cpu_features.cmake
@@ -119,6 +119,22 @@ elseif (ARCH_AMD64)
         SET(ENABLE_AVX512_FOR_SPEC_OP 0)
     endif()
 
+    # Same quick-n-dirty check for x86 as above for ARM, see above.
+    if (OS_LINUX AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64|x86_64" AND NOT NO_SSE3_OR_HIGHER)
+        # Test for flags in standard profile but not in NO_SSE3_OR_HIGHER profile.
+        # /proc/cpuid for Intel Xeon 8124: "fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse
+        # sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology nonstop_tsc cpuid aperfmperf
+        # tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c
+        # rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx
+        # avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke""
+        execute_process(
+            COMMAND grep -P "^(?=.*ssse3)(?=.*sse4_1)(?=.*sse4_2)" /proc/cpuinfo
+            OUTPUT_VARIABLE FLAGS)
+        if (NOT FLAGS)
+            MESSAGE(FATAL_ERROR "The build machine does not satisfy the minimum CPU requirements, try to run cmake with -DNO_SSE3_OR_HIGHER=1")
+        endif()
+    endif()
+
     # ClickHouse can be cross-compiled (e.g. on an ARM host for x86) but it is also possible to build ClickHouse on x86 w/o AVX for x86 w/
     # AVX. We only check that the compiler can emit certain SIMD instructions, we don't care if the host system is able to run the binary.
     # Therefore, use check_cxx_source_compiles (= does the code compile+link?) instead of check_cxx_source_runs (= does the code

--- a/cmake/cpu_features.cmake
+++ b/cmake/cpu_features.cmake
@@ -60,6 +60,23 @@ elseif (ARCH_AARCH64)
         set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument")
     endif ()
 
+    # Quick-n-dirty check: The build generates and executes intermediate binaries, e.g. protoc and llvm-tablegen. If we build on ARM for ARM
+    # and the build machine is too old, i.e. doesn't satisfy above modern profile, then these intermediate binaries will not run (dump
+    # SIGILL). Even if they could run, the build machine wouldn't be able to run the ClickHouse binary. In that case, suggest to run the
+    # build with the compat profile.
+    if (OS_LINUX AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)" AND NOT NO_ARMV81_OR_HIGHER)
+        # CPU features in /proc/cpuinfo and compiler flags don't align :( ... pick some obvious flags contained in modern but not legacy
+        # profile (full Graviton 3 /proc/cpuinfo is "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt
+        # fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs paca pacg dcpodp svei8mm svebf16 i8mm bf16
+        # dgh rng")
+        execute_process(
+            COMMAND grep -P "^(?=.*atomic)(?=.*ssbs)" /proc/cpuinfo
+            OUTPUT_VARIABLE FLAGS)
+        if (NOT FLAGS)
+            MESSAGE(FATAL_ERROR "The build machine does not satisfy the minimum CPU requirements, try to run cmake with -DNO_ARMV81_OR_HIGHER=1")
+        endif()
+    endif()
+
 elseif (ARCH_PPC64LE)
     # By Default, build for power8 and up, allow building for power9 and up
     # Note that gcc and clang have support for x86 SSE2 intrinsics when building for PowerPC

